service/src/main/java/tds/session/web/endpoints/SessionController.java

```diff
@@ -64,6 +65,12 @@ public SessionController(SessionService sessionService) {
         return new ResponseEntity<>(response, headers, HttpStatus.OK);
     }
 
+    @RequestMapping(value = "/{sessionId}/extend", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    void extendSession(@PathVariable final UUID sessionId) {
+        sessionService.updateDateVisited(sessionId);
```

Comment:
If this is called with an invalid sessionId (not present) then a 500 is thrown.  I think it should be a 400 which means we need to verify the session exists.  Considering it is going to be cached I don't think it will be a huge performance hit.

Unless there is a lower level Exception you can catch in the repository that would result in a NotFound being thrown.